File: backend/app/models/meeting_room.py
from sqlalchemy import Boolean, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class MeetingRoom(Base):
    """
    会議室情報を管理するDBモデル。

    予約対象となる会議室の名称、定員、有効状態を管理する。
    """

    __tablename__ = "meeting_rooms"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        index=True,
    )
    name: Mapped[str] = mapped_column(
        String(100),
        unique=True,
        index=True,
        nullable=False,
    )
    capacity: Mapped[int] = mapped_column(
        Integer,
        nullable=False,
    )
    location: Mapped[str] = mapped_column(
        String(255),
        nullable=False,
    )
    is_active: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False,
        default=True,
    )

File: frontend/src/types/meetingRoom.ts
export type MeetingRoom = {
  id: number;
  name: string;
  capacity: number;
  location: string;
  is_active: boolean;
};

export type MeetingRoomCreateRequest = {
  name: string;
  capacity: number;
  location: string;
};

export type MeetingRoomUpdateRequest = {
  name: string;
  capacity: number;
  location: string;
  is_active: boolean;
};